### DISCUSSION TITLE
how do I read grammar?

### DISCUSSION THREAD

**Discussion by aurkaxi** [rwf2/Rocket#2652] [Questions] [answered]
![image](https://github.com/rwf2/Rocket/assets/72933395/76800305-7b18-4d08-9f78-063e5e1a40be)
sorry but this grammar looks scary, i don't understand what each line or symbols even  mean.

i guess this indicates how do i write [#syntax] here. but this looks nothing like examples shown above. :)

**Comment by SergioBenitez** [comment, marked as answer]
The grammar indeed defines the valid syntax for the attribute. It's written in [EBNF](https://en.m.wikipedia.org/wiki/Extended_Backus–Naur_form), which is a common metasyntax (language) to more formally define the syntax of a language. It's certainly not necessary to understand the EBNF to use Rocket or fully understand how to use the attribute - the examples and rest of the documentation should suffice, and if it doesn't we would consider it a documentation bug. Instead, it's there to precisely define the syntax and serve as a reference and specification. In other words, to rigorously concretize the attribute.
